test/com/sun/tools/jmake/PCDEntryTest.java

```diff
@@ -0,0 +1,41 @@
+/* Copyright (c) 2015 Twitter, Inc. All rights reserved
+ *
+ * This program is distributed under the terms of
+ * the GNU General Public License Version 2. See the LICENSE file
+ * at the top of the source tree.
+ */
+package com.sun.tools.jmake;
+
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+public class PCDEntryTest {
+    private PCDEntry createPCDEntry(String className, String filename) {
+        return new PCDEntry(className,
+                            filename,
+                            filename.replace(".java", ".class"),
+                            1000000000000L,
+                            1000000000000L,
+                            null);
+
+    }
+
+    @Test
+    public void testIsPackagePrivateClass() {
+        assertEquals(false, createPCDEntry("foo/bar/Foo", "/tmp/foo/bar/Foo.java").isPackagePrivateClass());
+        assertEquals(false, createPCDEntry("foo/bar/Foo$1", "/tmp/foo/bar/Foo.java").isPackagePrivateClass());
+        assertEquals(true, createPCDEntry("foo/bar/Bar", "/tmp/foo/bar/Foo.java").isPackagePrivateClass());
```

Comment:
I'm not sure why Bar.Baz (Bar$1) wouldn't be.  I'm at the wrong computer
right now, so I can't actually run the test, but I think that's what it
would show, as well.

On Fri, Jan 23, 2015 at 9:34 AM, Eric Ayers <notifications@github.com>
wrote:

> In test/com/sun/tools/jmake/PCDEntryTest.java
> <https://github.com/pantsbuild/jmake/pull/14#discussion_r23452479>:
>
> > +public class PCDEntryTest {
> > +    private PCDEntry createPCDEntry(String className, String filename) {
> > +        return new PCDEntry(className,
> > +                            filename,
> > +                            filename.replace(".java", ".class"),
> > +                            1000000000000L,
> > +                            1000000000000L,
> > +                            null);
> > +
> > +    }
> > +
> > +    @Test
> > +    public void testIsPackagePrivateClass() {
> > +        assertEquals(false, createPCDEntry("foo/bar/Foo", "/tmp/foo/bar/Foo.java").isPackagePrivateClass());
> > +        assertEquals(false, createPCDEntry("foo/bar/Foo$1", "/tmp/foo/bar/Foo.java").isPackagePrivateClass());
> > +        assertEquals(true, createPCDEntry("foo/bar/Bar", "/tmp/foo/bar/Foo.java").isPackagePrivateClass());
>
> given this example, I'm not sure what a package private class really is
> supposed to be.
>
> Seems like in this case:
>
> Foo.java
>
> public class Foo {
> }
>
> class Bar {
>    class Baz {
>    }
> }
>
> that Bar would be considered package private, but Bar.Baz wouldn't be?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pantsbuild/jmake/pull/14/files#r23452479>.
>